spec/factories/assignments.rb

```diff
@@ -1,6 +1,9 @@
 FactoryBot.define do
   factory :assignment do
     user factory: %i[caseworker]
-    claim
+    transient do
+      claim { nil }
+    end
+    crime_application { claim || create(:claim) }
```

Comment:
You're right, that was messy! I was bending over backwards a bit to please rubocop, which doesn't like `a { create(:b) }` in factories, but then I realised I was really just trying to avoid updating old test code that assumed that assignments had a `claim` attribute. There was only 1 place that actually needed updating, so when I did that, I was left with the nice and straightforward:

```ruby
 submission factory: :claim
 ```
 
which avoids the need for a transient at all.